### PR TITLE
GCW-3301- Selecting default printer while doing `print_label` command

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3301

### What does this PR do?

This PR accepts `tag` params from app and print label on favourite printer as per tag params.

### CODE REVIEW

**Review by swatijadhav — APPROVED** [review]

**Review by shreyas098 — APPROVED** [review]

File: app/jobs/print_label_job.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-  def perform(package_id, current_user_id, label_type, print_count)
+  def perform(package_id, current_user_id, opts={})
     package = Package.find(package_id)
-    options = { inventory_number: package.inventory_number, print_count: print_count }
-    label = "Labels::#{label_type.classify}".safe_constantize.new(options)
-    printer = User.find_by_id(current_user_id).try(:printers).first
+    options = { inventory_number: package.inventory_number, print_count: opts[:print_count] }
+    label = "Labels::#{opts[:label_type].classify}".safe_constantize.new(options)
+    printer = PrintersUser.where(user_id: current_user_id, tag: opts[:tag]).first.try(:printer)
     PrintLabel.new(printer, label).print
```
Comment (patrixr) [inline app/jobs/print_label_job.rb:7]: this is at risk of breaking since printer can be nil

File: app/controllers/api/v1/packages_controller.rb [hunk 2]
```diff
@@ -212,7 +212,12 @@ def package_valuation
       end
 
       def print_inventory_label
-        PrintLabelJob.perform_later(@package.id, User.current_user.id, "inventory_label", print_count)
+        opts = {
```
Comment (patrixr) [inline app/controllers/api/v1/packages_controller.rb:215]: It's probably better to lookup the printer at this step. That way if the printer isn't set for this tag, you can return an error.

Comment (abulsayyad123) [reply app/controllers/api/v1/packages_controller.rb:215]: ohh.. thanks for pointing it out, will add check for this.

File: app/controllers/api/v1/packages_controller.rb [hunk 3]
```diff
@@ -212,7 +212,13 @@ def package_valuation
       end
 
       def print_inventory_label
-        PrintLabelJob.perform_later(@package.id, User.current_user.id, "inventory_label", print_count)
+        printer = PrintersUser.where(user_id: User.current_user.id, tag: params[:tag]).first.try(:printer)
```
Comment (bharat619) [inline app/controllers/api/v1/packages_controller.rb:215]:
```suggestion
        printer = PrintersUser.where(user_id: current_user.id, tag: params[:tag]).first.try(:printer)
```

File: app/controllers/api/v1/packages_controller.rb [hunk 4]
```diff
@@ -212,7 +212,13 @@ def package_valuation
...
+        return render json: { errors: I18n.t("package.printer_not_found") }, status: 400 unless printer
+        opts = {
+          print_count: print_count,
+          label: "inventory_label"
+        }
+        PrintLabelJob.perform_later(@package.id, printer.id, opts)
```
Comment (bharat619) [inline app/controllers/api/v1/packages_controller.rb:221]: There is already an existing `print_inventory_label` method in the `packages_controller.rb`, probably you can enhance that to take in arguments

or even better move it to a model method
```
# from controller
printer.print_inventory_label(opts)

# in model 
def print_inventory_label
 PrintLabelJob.perform_later(slef.id, opts[:user_id], "inventory_label", opts[:print_count])
end
```

**Review by bharat619 — APPROVED** [review]

**Review by patrixr — APPROVED** [review]

**Pull request merged**